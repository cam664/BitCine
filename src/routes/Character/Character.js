import React from 'react';
import StarwarsCharacter from '../../features/StarwarsCharacter';

const Character = () => {
  return (
    <main>
      <StarwarsCharacter />
    </main>
  );
}

export default Character;