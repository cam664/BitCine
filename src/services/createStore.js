import { createStore, combineReducers, applyMiddleware } from "redux";
import thunkMiddleware from "redux-thunk";

import { StarwarsListReducer } from '../features/StarwarsList';

const reducer = combineReducers({
  StarwarsList: StarwarsListReducer,
});

export default (initialState) => 
  createStore(
    reducer, 
    initialState, 
    applyMiddleware(thunkMiddleware)
  );
