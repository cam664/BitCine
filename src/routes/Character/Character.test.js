import React from 'react';
import renderer from 'react-test-renderer';

import Character from './Character';
import { Provider as ReduxProvider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import createStore from '../../services/createStore';

const store = createStore(window.__PRELOADED_STATE__);

describe('Character Route', () => {
  it('renders a snapshot', () => {
    const component = renderer.create(
      <ReduxProvider store={store}>
        <Router>
          <Character />
        </Router>
      </ReduxProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});