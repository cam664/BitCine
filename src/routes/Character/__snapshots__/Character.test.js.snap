// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Route renders a snapshot 1`] = `
<main>
  <img
    className="css-14hden7-Spinner"
    src="spinner.gif"
  />
</main>
`;
